pkg/dao/repositories.go

```diff
@@ -72,17 +72,27 @@ func (p repositoryDaoImpl) FetchForUrl(url string) (Repository, error) {
 	return internalRepo, nil
 }
 
-func (p repositoryDaoImpl) List(ignoreFailed bool) ([]Repository, error) {
+func (p repositoryDaoImpl) ListForIntrospection(urls *[]string, force bool) ([]Repository, error) {
 	var dbRepos []models.Repository
 	var repos []Repository
 	var repo Repository
-	var result *gorm.DB
 
-	if ignoreFailed {
-		result = p.db.Where("failed_introspections_count < ?", config.FailedIntrospectionsLimit+1).Find(&dbRepos)
-	} else {
-		result = p.db.Find(&dbRepos)
+	db := p.db
+	if !force && !config.Get().Options.AlwaysRunCronTasks {
+		introspectThreshold := time.Now().Add(config.IntrospectTimeInterval * -1) // Add a negative duration
+		db = db.Where(
+			db.Where("status != ?", config.StatusValid).
+				Or("last_introspection_time is NULL").                   // It was never introspected
+				Or("last_introspection_time < ?", introspectThreshold)). // It was introspected more than the threshold ago)
+			Where(
+				db.Where("failed_introspections_count < ?", config.FailedIntrospectionsLimit).
```

Comment:
I think this should be `config.FailedIntrospectionsLimit+1`, as it was before. I did that so the error on the repository would be "failed because it's over the limit", before introspection started skipping. This way the repository being skipped is communicated in the UI.

Reply:
Otherwise I think this would just silently start skipping introspection for repository over the limit